### ISSUE TITLE
Error when building: Unexpected token u in JSON at position 0

### ISSUE THREAD

**Issue by rijk** [vercel/next.js#45515]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
npx next info

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.2.0: Fri Nov 11 02:06:26 PST 2022; root:xnu-8792.61.2~4/RELEASE_ARM64_T8112
    Binaries:
      Node: 16.18.1
      npm: 8.19.2
      Yarn: N/A
      pnpm: 7.26.3
    Relevant packages:
      next: 13.1.7-canary.2
      eslint-config-next: 13.1.1
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

https://github.com/rijk/repro-rsc-issue/tree/repro

### To Reproduce

Run `pnpm run build`

### Describe the Bug

I was tracking down another issue with Vercel support (requesting a certain page with RSC header returned an error 500). After updating my reproduction to Next canary it no longer wanted to build at all:

```
[==  ] info  - Generating static pages (4/13)SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
```

Support engineer asked me to create an issue for it.

### Expected Behavior

No error.

### Which browser are you using? (if relevant)

Not relevant

### How are you deploying your application? (if relevant)

Vercel

**Comment by rijk** [comment]
All right, I've discovered this is caused by a conditional return in several head components. Adding `return null` to the end fixed the build issue.

As an aside (and why I'll leave this open although the original issue is fixed), while debugging I found that if a page does not have a `generateStaticParams()` (or returns `[]` from it), it is called with params like this:

```js
{
  organization: '%5Borganization%5D',
  path: '%5Bpath%5D',
  lesson: '%5Blesson%5D'
}
```

Is this intentional? It strikes me as odd.
